uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Form/WorkPattern.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+require_once 'CRM/Core/Form.php';
+
+/**
+ * Form controller class
+ *
+ * @see http://wiki.civicrm.org/confluence/display/CRMDOC43/QuickForm+Reference
+ */
+class CRM_HRLeaveAndAbsences_Form_WorkPattern extends CRM_Core_Form
+{
+
+    /**
+     * An array used to store the loaded WorkPattern's default values, so we
+     * only need to load them once.
+     *
+     * @var array
+     */
+    private $defaultValues = [];
+
+    public function setDefaultValues() {
+        if(empty($this->defaultValues)) {
+            if ($this->_id) {
+                $this->defaultValues = CRM_HRLeaveAndAbsences_BAO_WorkPattern::getDefaultValues($this->_id);
+            } else {
+                $this->defaultValues = [
+                    'is_active' => 1
+                ];
+            }
+        }
+
+        return $this->defaultValues;
+    }
+
+    public function buildQuickForm()
+    {
+        $this->_id = CRM_Utils_Request::retrieve('id' , 'Positive', $this);
```

Comment:
so the _id is a dynamically created property ... why not to define it above in the class  ?..